test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,314 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+                  type: "object",
+                  enum: [[1,3], 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+
+  it("should return an error for a null value in a enum object type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "objectSample",
+                in: "query",
+                schema: {
+                  type: "object",
+                  enum: [null]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+  
+  it("should not return an error when all items are array in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "objectSample",
+                in: "query",
+                schema: {
+                  type: "object",
+                  enum: [{ok: "Sample"},{}]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Boolean Tests
+
+  it("should return an error for a non boolean value in a boolean array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "booleanEnum",
+                in: "query",
+                schema: {
+                  type: "boolean",
+                  enum: [1, true, false]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
```

Comment:
**Change request**: add an expectation for the path of the error as well

Reply:
Expectation for path added, and Include pattern implemented.